[nmstate-1.4] nm: Do not touch interface DNS if global DNS is used

cathay4t opened the pull request:
When desire state only contains DNS configurations, we noticed nmstate
also purging interface DNS even global DNS is chose, this lead to
unnecessary modification of existing interfaces.

This patch skip the interface DNS modification if global DNS is used.

Integration test case included.

Resolves: https://issues.redhat.com/browse/RHEL-32218

cathay4t committed:
rpm SPEC: Use cargo vendor

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
Has CI failure. Still coding.

cathay4t committed:
nm: Do not touch interface DNS if global DNS is used

When desire state only contains DNS configurations, we noticed nmstate
also purging interface DNS even global DNS is chose, this lead to
unnecessary modification of existing interfaces.

This patch skip the interface DNS modification if global DNS is used
unless user is purging DNS configurations.

Integration test case included.

Resolves: https://issues.redhat.com/browse/RHEL-32218

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
CI: Install teamd before start the tests

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
CI: Skip vxlan tests in Github Action CI

Due to nispor bug on querying vxlan on new kernel(Fedora 38+ or Ubuntu
20.04):
https://issues.redhat.com/browse/RHEL-5001

nmstate will fail the verification on vxlan information missing.

The nmstate-1.4 branch is targeting the maintain for RHEL 8, OS with new
kernel should use nmstate 2.x releases.

Hence skip all vxlan integration tests in github action CI.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
CI: Install NetworkManager-config-server

The 1.4 branch test still assume the exist of
`NetworkManager-config-server`, instead of patch test cases, we just
install it back and restart NM daemon.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
Ready for review.

liangwen12year commented on .github/workflows/run_test.sh:
```diff
@@ -26,12 +26,12 @@ if [ $OS_TYPE == "c8s" ];then
     CUSTOMIZE_ARG='--customize=
         dnf install -y python3-varlink libvarlink-util python3-jsonschema \
-            python3-nispor;
+            python3-nispor teamd NetworkManager-team;
```
Restart NetworkManager ?

cathay4t replied:
The `start_machine_services()` will restart the daemon

liangwen12year commented on automation/tests-container-utils.sh:
```diff
@@ -70,10 +70,6 @@ function container_pre_test_setup {
 
-    if [[ "$CI" == "true" ]];then
-        rebuild_container_images
-    fi
```
Can you explain in the commit message what is the reason not building the container image in the 1.4 branch ?

cathay4t replied:
Done.

liangwen12year commented on packaging/nmstate.spec:
```diff
@@ -84,6 +85,7 @@ This package contains the nmstate plugin for OVS database manipulation.
 
 %prep
 %setup -q
+%cargo_prep -V 1
```
is this the correct syntax for preparing the project for building  ?https://tmz.fedorapeople.org/guidelines/packaging-guidelines/Rust/#_others

cathay4t replied:
This is for RHEL 8 where they do not have Fedora-like fancy macros.

cathay4t replied:
Oh. `-V` is for vendoring in RHEL.

cathay4t committed:
CI: Do not rebuild container image in 1.4 branch

The `rebuild_container_images()` is comparing docker file with main/base
branch(2.x) which has dropped stuff required for 1.4. There is not point
of doing new builds of CI container for 1.4 branch any more.

Signed-off-by: Gris Ge <fge@redhat.com>

liangwen12year commented:
The CI failure `pull-nmstate-integ_tier1-k8s` seems not related to the patch though.

cathay4t commented:
k8s CI is not maintained anymore for nmstate 1.4 branch and we cannot remove it. Please ignore its failure.
